include/Engine/Rendering/SpriteJob.h

```diff
@@ -48,6 +48,24 @@ struct SpriteJob : RenderJob
 
         FillColor = fillColor;
         FillPercentage = fillPercentage;
+
+        glm::vec3 scale = Transform::AbsoluteScale(world, cSprite.EntityID);
+
+        if((bool)cSprite["KeepRatioX"] == true) {
+            ScaleX = scale.x;
+        }
+        if ((bool)cSprite["KeepRatioY"] == true) {
+            ScaleY = scale.y;
+        }
+        if((bool)cSprite["KeepRatio"] == true) {
+            if(scale.y >= scale.x) {
+                ScaleY = (scale.x)/(scale.y);
+                ScaleX = 1;
+            } else {
+                ScaleY = 1;
+                ScaleX = (scale.x)/(scale.y);
```

Comment:
No, the Keepratio will override the other 2, it's basicly ratioX, ratioY or ratioBoth